src/stan/lang/ast_def.cpp

```diff
@@ -284,6 +284,16 @@ namespace stan {
     void function_signatures::add_unary(const::std::string& name) {
       add(name, DOUBLE_T, DOUBLE_T);
     }
+    void function_signatures::add_unary_vectorized(const::std::string&
+    name) {
+      for (size_t i = 0; i < 8; ++i) {
+        add(name, expr_type(DOUBLE_T, i), expr_type(INT_T, i));
```

Comment:
this will promote all the function outputs for `int` inputs. Are there any cases where the output is `int` with an `int` input? It seems like there would be.